src/transformers/models/longt5/configuration_longt5.py

```diff
@@ -0,0 +1,159 @@
+            The epsilon used by the layer normalization layers.
+        initializer_factor (`float`, *optional*, defaults to 1):
+            A factor for initializing all weight matrices (should be kept to 1, used internally for initialization
+            testing).
+        feed_forward_proj (`string`, *optional*, defaults to `"relu"`):
+            Type of feed forward layer to be used. Should be one of `"relu"` or `"gated-gelu"`. LongT5v1.1 uses the
+            `"gated-gelu"` feed forward projection. Original LongT5 implementation uses `"gated-gelu"`.
+        encoder_attention_type (`string`, *optional*, defaults to `"local"`):
+            Type of encoder attention to be used. Should be one of `"local"` or `"transient-global"`, which are
+            supported by LongT5 implementation.
+        use_cache (`bool`, *optional*, defaults to `True`):
+            Whether or not the model should return the last key/values attentions (not used by all models).
+    """
+    model_type = "longt5"
+    keys_to_ignore_at_inference = ["past_key_values"]
+    attribute_map = {"hidden_size": "d_model", "num_attention_heads": "num_heads", "num_hidden_layers": "num_layers"}
+
+    def __init__(
+        self,
+        vocab_size=32128,
+        d_model=512,
+        d_kv=64,
+        d_ff=2048,
+        num_layers=6,
+        num_decoder_layers=None,
+        num_heads=8,
+        local_radius=127,
+        global_block_size=16,
+        relative_attention_num_buckets=32,
+        relative_attention_max_distance=128,
+        dropout_rate=0.1,
+        layer_norm_epsilon=1e-6,
+        initializer_factor=1.0,
+        feed_forward_proj="relu",
+        is_encoder_decoder=True,
+        encoder_attention_type="local",
+        use_cache=True,
+        pad_token_id=0,
+        eos_token_id=1,
+        **kwargs
+    ):
+        self.vocab_size = vocab_size
+        self.d_model = d_model
+        self.d_kv = d_kv
+        self.d_ff = d_ff
+        self.num_layers = num_layers
+        self.num_decoder_layers = (
+            num_decoder_layers if num_decoder_layers is not None else self.num_layers
+        )  # default = symmetry
+        self.num_heads = num_heads
+        self.local_radius = local_radius
+        self.global_block_size = global_block_size
+        self.relative_attention_num_buckets = relative_attention_num_buckets
+        self.relative_attention_max_distance = relative_attention_max_distance
+        self.dropout_rate = dropout_rate
+        self.layer_norm_epsilon = layer_norm_epsilon
+        self.initializer_factor = initializer_factor
+        self.feed_forward_proj = feed_forward_proj
+        self.encoder_attention_type = encoder_attention_type
+        self.use_cache = use_cache
+        super().__init__(
+            pad_token_id=pad_token_id,
+            eos_token_id=eos_token_id,
+            is_encoder_decoder=is_encoder_decoder,
+            **kwargs,
+        )
```

Comment:
Currently, if this is moved to the top some tests are failing:
```
FAILED tests/longt5/test_modeling_longt5.py::LongT5ModelTest::test_config - AssertionError: 512 != 0 : `hidden_size value 0 expected, but was 512
FAILED tests/longt5/test_modeling_longt5.py::LongT5TGlobalModelTest::test_config - AssertionError: 512 != 0 : `hidden_size value 0 expected, but was 512
FAILED tests/longt5/test_modeling_longt5.py::LongT5EncoderOnlyModelTest::test_config - AssertionError: 512 != 0 : `hidden_size value 0 expected, but was 512
FAILED tests/longt5/test_modeling_longt5.py::LongT5EncoderOnlyTGlobalModelTest::test_config - AssertionError: 512 != 0 : `hidden_size value 0 expected, but was 512
```

due to the following mapping.
```python
attribute_map = {"hidden_size": "d_model", "num_attention_heads": "num_heads", "num_hidden_layers": "num_layers"}
```

Is it reasonable to replace the old T5-like names with the standard `transformers` naming?

Reply:
Interesting! Could you maybe still move it to the top and then we'll look into fixing the failing tests afterward? This does seem to be a bug

Reply:
Actually @sgugger we do the same for `configuration_t5.py` I guess because of the `attribute_map`: https://github.com/huggingface/transformers/blob/3a71e94a929fae8c04bdad9129c461c920413a2d/src/transformers/models/t5/configuration_t5.py#L119

@nreimers was this done when we've added the attribute maps?

Reply:
Ok for me then!